fastplotlib/layouts/_plot_area.py

```diff
@@ -428,27 +569,37 @@ def clear_animations(self, removal: str = None):
             The type of animation functions to clear. One of 'pre' or 'post'. If `None`, removes all animation
             functions.
         """
+        to_remove = list()
+
         if removal is None:
             # remove all
             for func in self._animate_funcs_pre:
-                self._animate_funcs_pre.remove(func)
+                to_remove.append(func)
 
             for func in self._animate_funcs_post:
-                self._animate_funcs_post.remove(func)
+                to_remove.append(func)
+
         elif removal == "pre":
             # only pre
             for func in self._animate_funcs_pre:
-                self._animate_funcs_pre.remove(func)
+                to_remove.append(func)
+
         elif removal == "post":
             # only post
             for func in self._animate_funcs_post:
-                self._animate_funcs_post.remove(func)
+                to_remove.append(func)
         else:
             raise ValueError(
                 f"Animation type: {removal} must be one of 'pre' or 'post'. To remove all animation "
                 f"functions, pass `type=None`"
             )
 
+        for func in to_remove:
+            if func in self._animate_funcs_persist:
+                # skip
+                continue
+            self.remove_animation(func)
+
```

Comment:
I need to get rid of this `persist` thing. It was an idea for updating the tooltip on every render cycle when I implemented tooltips being managed by a `Graphic` and is not required anymore.